core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -892,3 +871,219 @@ object ZkPath {
     client.createPersistentSequential(path, data)
   }
 }
+
+/**
+ * Implements an ephemeral znode that is recreated in the case it disappears.
+ * A znode can be deleted in this case if it has been created in a previous session.
```

Comment:
"A znode ..." I feel the comments here are not very clear, my understanding is that:

"Upon session expiration, the ephemeral node created from the session will be eventually deleted, and the new session trying to re-create the node will fail when it is not yet deleted by ZooKeeper, in this case it will just add a watcher on the path which will re-try creating the node when being notified that the node from the previous session has been deleted".